src/util.jl

```diff
@@ -79,12 +80,34 @@ function _loadtable_serial(T, file::Union{IO, AbstractString, AbstractArray};
         samecols = map(x->map(string, x), samecols)
     end
 
-    if isa(file, AbstractArray)
-        cols, header, count = csvread(file, delim;
-                                      samecols=samecols,
-                                      kwargs...)
+    if file isa Tuple
+        _file, bidx, header = file
+        # TODO: Make this more efficient in block-io.jl
+        fsize = filesize(_file)
+        nblocks = max(div(fsize, blocksize), 1)
+        bios = blocks(_file, '\n', nblocks)
```

Comment:
every call to `blocks` will read the whole file character-by-character! So this is pretty redundant. The major difficulty here is that we cannot transfer IO types across processors. But we can transfer offset and length for each block. So it would be nice to have the offset and length for each block instead of just the `idx` in the tuple.

Reply:
So maybe the BlockIO constructor should be split into two stages: one that computes the offsets, another that returns the BlockIO.

Reply:
Wow, I must have been really desperate to get this PR working that I didn't even consider how inefficient this was! I'll get this fixed.